docs/examples.md

```diff
@@ -5,87 +5,88 @@ of pull streams are used in simple real-world examples.
-it becomes much more likely that the sink errors (the network connection
+response, because that data is already at hand. When the old data is all
+written then the output rate may drop significantly because the server (the source) will
+wait for new data to be added to the file. Therefore,
+it becomes much more likely that the sink will error (for example if the network connection
 drops) while the source is waiting for new data. Because of this,
 it's necessary to be able to abort the stream reading (after you called
 read, but before it called back). If it was not possible to abort
 out of turn, you'd have to wait for the next read before you can abort
-but, depending on the source of the stream, that may never come.
+but, depending on the source of the stream, the next read may never come.
 
-# a through stream that needs to abort.
+# A through stream that needs to abort.
 
-Say we read from a file (source), JSON parse each line (through),
+Say we wish to read from a file (source), parse each line as JSON (through),
 and then output to another file (sink).
-because there is valid and invalid JSON, the parse could error,
-if this parsing is a fatal error, then we are aborting the pipeline
-from the middle. Here the source is normal, but then the through fails.
-When the through finds an invalid line, it should abort the source,
+If the parser encounters illegal JSON then it will error and,
+if this parsing is a fatal error, then the parser needs to abort the pipeline
+from the middle. Here the source reads normaly, but then the through fails.
+When the through finds an invalid line, it should first abort the source,
 and then callback to the sink with an error. This way,
 by the time the sink receives the error, the entire stream has been cleaned up.
 
-(you could abort the source, and error back to the sink in parallel,
-but if something happened to the source while aborting, for the user
-to know they'd have to give another callback to the source, this would
-get called very rarely so users would be inclined to not handle that.
-better to have one callback at the sink.)
-
-In some cases you may want the stream to continue, and just ignore
-an invalid line if it does not parse. An example where you definately
-want to abort if it's invalid would be an encrypted stream, which
+(You could abort the source and error back to the sink in parallel.
+However, if something happened to the source while aborting, for the user
+discover this error they would have to call the source again with another callback, as
+situation would occur only rarely users would be inclined to not handle it leading to
+the possiblity of undetected errors.
+Therefore, as it is better to have one callback at the sink, wait until the source
+has finished cleaning up before callingback to the pink with an error.)
+
+In some cases you may want the stream to continue, and the the through stream can just ignore
+an any linesthat do not parse. An example where you definately
```

Comment:
Should be just "any lines that do parse."

Reply:
yep